src/gc/gc.d

```diff
@@ -432,6 +484,8 @@ struct GC
                 p = sentinel_sub(p);
                 auto biti = cast(size_t)(p - pool.baseAddr) >> pool.shiftBy;
 
+                //Based on the documetnation, shouldn't we set the
+                //bits first and then get the new value of the bit field?
```

Comment:
I probably should have mentioned this when I opened the request, but my intent here was to bring awareness to these various bits of code for the reasons stated in the questions.

I was planning on removing these before anything was actually committed, but now that I think about it it should probably have been brought up in a different PR or something.